network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -725,13 +780,19 @@ func (c *ControlMsgValidationInspector) logAndDistributeAsyncInspectErrs(req *In
 	case IsErrUnstakedPeer(err):
 		lg.Warn().Err(err).Msg("control message received from unstaked peer")
 	default:
-		err = c.distributor.Distribute(p2p.NewInvalidControlMessageNotification(req.Peer, ctlMsgType, err))
+		notification := p2p.NewInvalidControlMessageNotification(req.Peer, ctlMsgType, err, count)
+		lg.Error().Err(notification.Error).
+			Str("control_msg_type", notification.MsgType.String()).
+			Uint64("err_count", count).
+			Msg("rpc control message async inspection failed")
+		err = c.distributor.Distribute(notification)
 		if err != nil {
 			lg.Error().
```

Comment:
According to the documentation of `Distribute` method, this should be `fatal`:
```go
	// Distribute distributes the event to all the consumers.
	// Any error returned by the distributor is non-recoverable and will cause the node to crash.
	// Implementation must be concurrency safe, and non-blocking.
```

Reply:
https://github.com/onflow/flow-go/pull/4773/commits/d43678575506a2192c400799ed67e8568c95ab07